files/__addonLocation__/gitignore

```diff
@@ -3,8 +3,3 @@
 # will also appear in published NPM packages.
 /README.md
 /LICENSE.md
-# Build output
```

Comment:
this isn't redundant because they are needed when using the `--addon-only` flag.

If anything, the entries from the root gitignore should be removed.

Reply:
> this isn't redundant because they are needed when using the --addon-only flag.

Well, but then I think we should make changes based on the fact that --addon-only was passed. The default output now has `/dist` ignored in both .gitignore files, which is redundant!?

> If anything, the entries from the root gitignore should be removed.

Why? They contain other stuff that needs to be ignored, like `node_modules/` or `.eslintcache`. 

Do you want to have like complete isolation of every workspace, where it defines its own config without relying on a global one? If that's you intent, then ok, that's a point you can make (I don't think I would agree tho). But then what we have _now_ as the local .gitignore covers by far not all the things that the addon folder needs to ignore, so it does already rely on the global (root) config.

If you look at e.g. https://github.com/CrowdStrike/ember-headless-form, is there anything wrong with that setup? It has multiple package, and each of their `/dist` and whatnot gets ignored by the global config.

Again, if you concern is only about the --addon-only flag, that we can make the output be specific to what we want to have in that context. But the default output should be reasonable, and I think it's confusing right now...

Reply:
Workspace isolation is my main motivation here' yeah. Drag and drop refactoring (even between repos) aligns with how we ane wanting to arrange content within our apps and addons, and vorkspace isolated configs enable that.

> Headless form

Looks perfectly fine -- Everyone can do their own thing! 

> --addon-only flag, that we can make the output be specific to what we want to have in that context

it's just more logic for us to maintain, is part of what i'm worried about, and with local-confics, they work in all scenario, so we have less if thens to do

> But the default output should be reasonable

That's a little harsh -- does this mean package isolation goals are unreasonable 😅

Reply:
Ok, if workspace isolation is your concern, that's valid. But that's not what we have _now_. Our blueprint does not try to achieve that, that's at least how I would see that...

The addons [.gitignore](https://github.com/embroider-build/addon-blueprint/blob/main/files/__addonLocation__/gitignore) duplicates(!) `/dist`, but still relies on a number of important entries like `node_modules` or `.eslintcache` from the [root .gitignore](https://github.com/embroider-build/addon-blueprint/blob/main/files/gitignore).

Compare that to the [test-app's .gitiignore](https://github.com/ember-cli/ember-cli/blob/master/blueprints/app/files/gitignore), which is "complete".

So _currently_ our blueprint does not achieve workspace isolation, I think that's a fact we can agree on?
But it does introduce redundancy, which is what I wanted to remove in this PR.

> That's a little harsh -- does this mean package isolation goals are unreasonable 😅

No, that's not what I was trying to say, sorry if it came across like that. As I said above, what I think is unreasonable is to have a blueprint that is inconsistent in what it does. It does currently a mix of having a global .gitignore and a bit of local, but incomplete config, leading to redundancy. It does not support workspace isolation _properly_ I think.

If we want workspace isolation, then I think we need a "Introduce workspace isolation" PR that does this properly. I'm still a bit on the fence if I want that, but I can be convinced! :wink:

Nevertheless I think this PR is still ok, as it is based on the fact that currently the blueprint does not claim to do complete workspace isolation (wrt to .gitignore at least), so it stays at that and just removes the recently introduced duplication. Which does not prevent us to land proper workspace isolation later. Does that make sense?

Reply:
makes sense -- I will open a separate PR "sometime" lol